bot/seasons/evergreen/TriviaQuiz.py

```diff
@@ -0,0 +1,263 @@
+        self.games = {}  # channel as key and value as instinct of dataclass GameData
+        self.categories = {
+            "retro": "Questions related to retro gaming."
+        }
+        self.inactivity_limit = 3  # Number of questions unanswered in a row after which quiz stops.
+
+    @staticmethod
+    def load_questions():
+        """Load the questions from  json file."""
+        p = Path("bot", "resources", "evergreen", "trivia_quiz.json ")
+        with p.open() as json_data:
+            questions = load(json_data)
+            return questions
+
+    @commands.group(name="tquiz", invoke_without_command=False)
+    async def tquiz(self, ctx):
+        """Trivia Quiz game for fun!"""
+        await ctx.send_help("tquiz")
+
+    @tquiz.command(name="start")
+    async def start(self, ctx, category=None):
+        """Start a quiz!
+
+        Questions for the quiz can be selected from the following categories:
+        - Retro : questions related to retro gaming.
+        """
+        await ctx.send("Quiz triggered! Gonna start in a couple of seconds...")
+        await asyncio.sleep(1)
+
+        # Checking if there is already a game running in that channel.
+        if ctx.channel.id in list(self.games.keys()):
+            return await ctx.send("Game already running in this channel!")
+
+        if category is None:
+            category = random.choice(list(self.categories.keys()))
+
+        else:
+            category = category.lower()
+            if category not in self.categories.keys():
+                embed = self.category_embed()
+                return await ctx.send(f"Category {category} does not exist!", embed=embed)
+
+        self.games[ctx.channel.id] = GameData(
+            owner=ctx.author.id,
+            category=category
+        )
+
+        await self.send_question(ctx.channel)
+
+    def category_embed(self):
+        """A function which returns an embed showing all avilable categories"""
+        embed = discord.Embed(colour=discord.Colour.blue())
+        embed.title = "The available question categories are:"
+        embed.description = ""
+        for cat, description in self.categories.items():
+            embed.description += f"**- {cat.capitalize()}**\n{description.capitalize()}\n"
+        embed.set_footer(text="If not category is chosen, then a random one will be selected.")
+        return embed
+
+    async def send_question(self, channel):
+        """This function is to be called whenever a question needs to be sent."""
+        await asyncio.sleep(2)
+        game = self.games[channel.id]
+        if game.unanswered_questions == self.inactivity_limit:
+            del self.games[channel.id]
+            return await channel.send("Game stopped due to inactivity.")
+
+        category = game.category
+        category_dict = self.questions[category]
+        question_dict = random.choice(category_dict)
+
+        same_question = True
+
+        # Making sure a question is not repeated.
```

Comment:
Rather than repeatedly calling `random.choice` when a repeated question is found, a better way would be to start with a list of shuffled questions, then only reshuffle when necessary.

Reply:
you mean make a list of shuffled questions first and then use one by one in order?